### PR TITLE
ci: cosmos-sdk nightly build

### PR DESCRIPTION
**Pull request by auricom**
Workflow is running daily and :
- Builds cosmos-sdk (through SimApp) for each  major-version declared in the matrix.
- Launch a testnet and checks that at least 4 blocks are produced.
- Upload the binary as workflow artifact
- Create a container image
- Test the container image using goss
- Publish the container image on ghcr (one tag for each major version, tag is multi-arch ready)

image

image

### TO-DO
- Provide a user for slack notifications

### CODE REVIEW

File: .github/workflows/release-cosmos-sdk.yaml [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+    needs: merge
+    runs-on: ubuntu-latest
+    if: ${{ failure() }}
+    steps:
+      - name: Notify Slack on failure
+        uses: rtCamp/action-slack-notify@v2.3.0
+        env:
+          SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK }}
+          SLACK_CHANNEL: sdk-sims
+          SLACK_USERNAME:
```
Comment (auricom) [inline .github/workflows/release-cosmos-sdk.yaml:324]: @julienrbrt not sure about what to use

Comment (julienrbrt) [reply .github/workflows/release-cosmos-sdk.yaml:324]: ditto

File: containers/cosmos-sdk/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,44 @@
+FROM cgr.dev/chainguard/wolfi-base:latest@sha256:0f1d81605bda6e2388c3c7f731700d8c12e17259d58ffba11f36ddc81d9c0a76
```
Comment (julienrbrt) [inline containers/cosmos-sdk/Dockerfile:1]: What's the difference between this and alpine? Why not alpine?

File: containers/cosmos-sdk/entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+set -x
+
+NODE_HOME=${NODE_HOME:-/config}
+START_CMD=${START_CMD:-start}
+
+if [[ ! -f "${NODE_HOME}/config.json" ]]; then
+    echo "Launch init procedure..."
+    /app/node config set client chain-id testchain --home ${NODE_HOME}
+    /app/node config set client keyring-backend test --home ${NODE_HOME}
+    cat /config/config/client.toml
```
Comment (julienrbrt) [inline containers/cosmos-sdk/entrypoint.sh:12]: can be deleted

**Review by julienrbrt — COMMENTED** [review]
Is the state persisted between two runs? Or you always run on an empty state?

**Comment by auricom** [comment]
> Is the state persisted between two runs? Or you always run on an empty state?

State is not persisted.

This workflow objective is to have battle-tested and production ready artifacts, that we are confident can be used to run long-term testnets (CI only).

I'm currently working on the follow-up, that will enable a continuous delivery experience to ensure that testnet will be deployed on target infrastructure : Hetzner VMs.
This target infra will be industrialized the same as we do with validators, and will have this time persistent data.

File: containers/cosmos-sdk/Dockerfile (hunk 2 above)
Comment (auricom) [reply containers/cosmos-sdk/Dockerfile:1]: I no longer recommend using alpine linux, because of compatibilty issues I had with DNS and glibc absence that may complexify deps management for building and runtime.

this well written blog post will sum it better than me : https://martinheinz.dev/blog/92

I attended a CNCF meeting a few months ago and learned about wolfi-os.
https://thenewstack.io/meet-wolfi-the-linux-distro-designed-to-shrink-your-supply-chain/

TLDR : 
- Better security (less CVEs, SBOMs)
- Lower footprint (cosmos-sdk image is ~35MB with a 25MB binary)
- No compatibility issues

On the cons side, project is less known (but migrating images to another base do not require rewrites) and there are less packages on the manager than alpine, but not very relevant for our use case imho.

File: .github/workflows/release-cosmos-sdk.yaml [hunk 4]
```diff
@@ -0,0 +1,328 @@
+name: Nightly builds of SimApp (Cosmos-SDK)
+on:
+  schedule:
+    - cron: 0 0 * * * # Every day at 0:00 UTC
+  workflow_dispatch:
+
```
Comment (tac0turtle) [inline .github/workflows/release-cosmos-sdk.yaml:6]: short comment of the all the things the action does would be useful. Allows people to avoid reading through all the steps

Comment (auricom) [reply .github/workflows/release-cosmos-sdk.yaml:12]: added !

**Review by julienrbrt — APPROVED** [review]
🚀
